elixir-phoenix-deployement.md

```diff
@@ -0,0 +1,162 @@
+is it is quick to setup and have your application running
+for your users.
+However in the longer term using a PaaS can become costly (per unit of compute e.g. RAM/CPU).
+Virtual Private Server are in contrast normally cheaper to run
+but require more setup time and DevOps knowledge to maintain.
+
+### PaaS
+
+#### Heroku
+
+We are used to deploy Elixir/Phoenix on Heroku
+and we have already a ["how to"](https://github.com/dwyl/learn-phoenix-framework/blob/master/heroku-deployment.md) guide of this process.
+
+However there are some limitations when using Heroku with Phoenix (see https://hexdocs.pm/phoenix/heroku.html):
+![image](https://user-images.githubusercontent.com/6057298/83642784-2e1d1200-a5a7-11ea-8c97-9c7dd920469c.png)
+
+pricing:
+![image](https://user-images.githubusercontent.com/6057298/83641712-cc0fdd00-a5a5-11ea-8cbf-49981ae3747e.png)
+
+The pricing above doesn't include the database which needs to be added to the total cost, see https://elements.heroku.com/addons/heroku-postgresql
+
+
+### Gigalixir
+
+Similar to Heroku [Gigalixir](https://www.gigalixir.com/) provides
+a platform which focuses _exclusively_ on deploying Elixir/Phoenix application.
+Having an exclusive focus on deploying Elixir Apps gives Gigalixir several key advantages:
+1. Zero-downtime blue-green continuous deployment.
+see: https://martinfowler.com/bliki/BlueGreenDeployment.html
+2. No limit to concurrent connections.
+3. All clustering handled transparently. 
+(so if you need to scale your app beyond 10k concurrent users, 
+you don't have to pay exponentially more for bigger "dynos" 
+the way you are forced to scale vertically on Heroku)
+
+Deploying: https://elixircasts.io/deploying-with-gigalixir-%28revised%29
+
+Pricing:
+
+![image](https://user-images.githubusercontent.com/6057298/83643629-290c9280-a5a8-11ea-89da-648daec204f3.png)
+
+Gigalixir vs Heroku:
+
+![image](https://user-images.githubusercontent.com/6057298/83644108-d1225b80-a5a8-11ea-869a-c8c0e4a28012.png)
+
+Like Heroku you need to add the cost for using the database:
+
+![image](https://user-images.githubusercontent.com/6057298/83644296-09c23500-a5a9-11ea-9fdd-c50532195c95.png)
+
+see also tiers pricing page: https://gigalixir.readthedocs.io/en/latest/tiers-pricing.html
+
+
+### Render
+
+[Render](https://render.com/) is another Paas similar to Heroku
+
+pricing:
+
+![image](https://user-images.githubusercontent.com/6057298/83645052-e9df4100-a5a9-11ea-8d3e-f1b18bb4cc02.png)
+
+
+
+## Virtual Private Server
+
+VPS allow us to manage ourself the deployement
+setup. This allow us to customise the server and
+the costs linked to it.
+
+
+### Linode
+
+#### Ubuntu
+
+Linode provides and support Ubuntu:
+![image](https://user-images.githubusercontent.com/6057298/83647811-3bd59600-a5ad-11ea-893c-b99e3df7f605.png)
+
+
+From there the idea is install Erlang and Elixir on the server
+and then to run the application.
+
+- Install Erlang/Elixir with asdf:
```

Comment:
@SimonLab as previously noted https://github.com/dwyl/phoenix-liveview-counter-tutorial/issues/17#issuecomment-625664043 please don't introduce a _completely_ new way of managing dependencies without giving _some_ context. This is the first mention of `asdf` in any file in the dwyl org. 

If you want to _propose_ using `asdf` https://github.com/asdf-vm/asdf as a version manager, and I have nothing against it in principal, it's as simple as opening an issue. see: https://github.com/dwyl/technology-stack/issues/18
Dropping a random tool into a doc without any context is not beginner friendly and sets an undesirable example to others who may think they can add _any_ dependency they like.